go-controller/pkg/ovn/controller/services/service_tracker.go

```diff
@@ -27,23 +26,24 @@ func splitVirtualIPKey(key string) (string, v1.Protocol) {
 
 // serviceTracker tracks the services VIPs using the service name and namespace as key
 // one service can have multiple VIPs, they are stored in the format IP:Port/Protocol
-// The services allows to map Kubernetes Services and OVN LoadBalancer
+// The services allows mapping between Kubernetes Services and OVN LoadBalancer
 type serviceTracker struct {
 	sync.Mutex
-	virtualIPByService map[string]sets.String
+	// holds a map of service mapping to map of vips and loadbalancers
+	virtualIPByService map[string]map[string]string
 	hadEndpoints       map[string]bool
 }
 
 // newServiceTracker creates and initializes a new serviceTracker.
 func newServiceTracker() *serviceTracker {
 	return &serviceTracker{
-		virtualIPByService: map[string]sets.String{},
+		virtualIPByService: map[string]map[string]string{},
```

Comment:
This double map is quite hard to follow.
I'd have the inner one described by a type with methods (which can still be baked by a map).